web/src/app/shared/components/type-ahead/type-ahead.component.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { DebugElement } from '@angular/core';
+import { AccessIdsService } from 'app/shared/services/access-ids/access-ids.service';
+import { TypeAheadComponent } from './type-ahead.component';
+import { BrowserModule, By } from '@angular/platform-browser';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { RouterModule } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { of } from 'rxjs/internal/observable/of';
+import { MatSelectModule } from '@angular/material/select';
+import { MatAutocompleteModule } from '@angular/material/autocomplete';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { FormsModule } from '@angular/forms';
+import { compileComponentFromMetadata, componentFactoryName } from '@angular/compiler';
+
+const AccessIdsServiceSpy = jest.fn().mockImplementation(
+  (): Partial<AccessIdsService> => ({
+    getAccessItems: jest.fn().mockReturnValue(of()),
+    searchForAccessId: jest.fn().mockReturnValue(of())
+  })
+);
+
+describe('TypeAheadComponent', () => {
+  let component: TypeAheadComponent;
+  let fixture: ComponentFixture<TypeAheadComponent>;
+  let debugElement: DebugElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [TypeAheadComponent],
+      imports: [
+        BrowserModule,
+        RouterModule,
+        RouterTestingModule,
+        HttpClientTestingModule,
+        MatSelectModule,
+        MatAutocompleteModule,
+        MatFormFieldModule,
+        MatInputModule,
+        FormsModule,
+        BrowserAnimationsModule
+      ],
+      providers: [{ provide: AccessIdsService, useClass: AccessIdsServiceSpy }]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TypeAheadComponent);
+    debugElement = fixture.debugElement;
+    component = fixture.debugElement.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should change value via the input field', async(() => {
+    component.value = 'val_1';
+    component.changeValue();
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
+      let input = fixture.debugElement.query(By.css('.typeahead__form-input'));
```

Comment:
A variable 'debugElement' was created in line 51. This variable should replace 'fixture.debugElement'.